src/Controller/FavoritesController.php

```diff
@@ -0,0 +1,189 @@
+<?php
+
+namespace Drupal\openy_gated_content\Controller;
+
+use Drupal\Core\Controller\ControllerBase;
+use Drupal\Core\Database\Connection;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+use Symfony\Component\HttpFoundation\JsonResponse;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\Serializer\Encoder\JsonEncoder;
+use Symfony\Component\Serializer\Serializer;
+
+/**
+ * Class Favorites Controller.
+ */
+class FavoritesController extends ControllerBase {
+
+  /**
+   * The current active database's master connection.
+   *
+   * @var \Drupal\Core\Database\Connection
+   */
+  protected $database;
+
+  /**
+   * The serializer.
+   *
+   * @var \Symfony\Component\Serializer\Serializer
+   */
+  protected $serializer;
+
+  /**
+   * Constructs a CustomFormattersController object.
+   *
+   * @param \Drupal\Core\Database\Connection $database
+   *   The current active database's master connection.
+   */
+  public function __construct(Connection $database) {
+    $this->database = $database;
+    $this->serializer = new Serializer([], [new JsonEncoder()]);
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container) {
+    return new static(
+      $container->get('database')
+    );
+  }
+
+  /**
+   * List of favorites items for current user.
+   *
+   * @return \Symfony\Component\HttpFoundation\JsonResponse
+   *   Json Array with favorites items grouped by entity type and bundle.
+   */
+  public function list() {
+    $data = [
+      'node' => [
+        'vy_blog_post' => [],
+        'gc_video' => [],
+      ],
+      'eventinstance' => [
+        'live_stream' => [],
+        'virtual_meeting' => [],
+      ],
+      'taxonomy_term' => [
+        'gc_category' => [],
+      ],
+    ];
+
+    $uid = $this->currentUser()->id();
+    $query = $this->database->select('vy_favorite_item', 'f');
```

Comment:
not cached function

Reply:
Fixed in https://github.com/fivejars/openy_gated_content/pull/231